source/common/http/access_log/access_log_formatter.cc

```diff
@@ -233,15 +239,23 @@ RequestInfoFormatter::RequestInfoFormatter(const std::string& field_name) {
     };
   } else if (field_name == "REQUEST_DURATION") {
     field_extractor_ = [](const RequestInfo& request_info) {
-      return std::to_string(std::chrono::duration_cast<std::chrono::milliseconds>(
-                                request_info.requestReceivedDuration())
-                                .count());
+      Optional<std::chrono::microseconds> duration = request_info.requestReceivedDuration();
+      if (duration.valid()) {
+        return std::to_string(
+            std::chrono::duration_cast<std::chrono::milliseconds>(duration.value()).count());
+      } else {
+        return UnspecifiedValueString;
+      }
     };
   } else if (field_name == "RESPONSE_DURATION") {
     field_extractor_ = [](const RequestInfo& request_info) {
-      return std::to_string(std::chrono::duration_cast<std::chrono::milliseconds>(
-                                request_info.responseReceivedDuration())
-                                .count());
+      Optional<std::chrono::microseconds> duration = request_info.responseReceivedDuration();
```

Comment:
When looking at this, I wondered about the relationship to https://github.com/envoyproxy/data-plane-api/blob/master/api/accesslog.proto and proto logging. It feels like `RequestInfo` is similar to `EnvoyAccessLog` in terms of what it represents at log generation time (although it's more stateful, and keeps track of stuff during requests). 

Would it makes sense to be building up a proto suitable for sending to gRPC access logs during HTTP/TCP requests, and then do a final pass to convert it to text if needed?

Reply:
I think this does make sense to see where we can consolidate. I can also take a look at this in the context of the streaming gRPC log handler work that I need to do.